libraries/AP_Mount/AP_Mount_Alexmos.cpp

```diff
@@ -26,72 +26,91 @@ void AP_Mount_Alexmos::update()
 
     read_incoming(); // read the incoming messages from the gimbal
 
+    // contains true if target is valid
+    bool isTargetValid = false;
+
     // update based on mount mode
     switch (get_mode()) {
         // move mount to a "retracted" position.  we do not implement a separate servo based retract mechanism
         case MAV_MOUNT_MODE_RETRACT: {
             const Vector3f &target = _params.retract_angles.get();
-            _angle_rad.roll = radians(target.x);
-            _angle_rad.pitch = radians(target.y);
-            _angle_rad.yaw = radians(target.z);
-            _angle_rad.yaw_is_ef = false;
+            mnt_target.target_type = MountTargetType::ANGLE;
+            mnt_target.angle_rad.roll = radians(target.x);
```

Comment:
Not a blocker but I was thinking that maybe we should add a "set()" method to the MountTarget structure (and turn it into a class) and then use it instead of specifying the members individually.